internal/api/server.go

```diff
@@ -33,6 +35,27 @@ func NewServer(store probestore.ProbeStorage) Server {
 	}
 }
 
+// validateProtectedLabels checks if the user is trying to modify protected system labels
+func validateProtectedLabels(labels *v1.LabelsSchema) error {
+	if labels == nil {
+		return nil
+	}
+
+	protectedLabels := []string{
+		baseAppLabelKey,      // "app"
+		probeStatusLabelKey,  // "rhobs-synthetics/status"
+		probeURLHashLabelKey, // "rhobs-synthetics/static-url-hash"
+	}
+
+	for _, protectedLabel := range protectedLabels {
+		if _, exists := (*labels)[protectedLabel]; exists {
```

Comment:
Any reason to make this a pointer? Syntax could be simplified a bit if we're able to just pass by value here